diff -bur poco-1.14.1-all-orig/Foundation/include/Poco/Platform.h poco-1.14.1-all/Foundation/include/Poco/Platform.h
--- poco-1.14.1-all-orig/Foundation/include/Poco/Platform.h	2025-03-01 17:44:37.631844300 -0700
+++ poco-1.14.1-all/Foundation/include/Poco/Platform.h	2025-03-01 17:45:02.693917300 -0700
@@ -250,6 +250,9 @@
 	#if __has_include(<cxxabi.h>)
 		#define POCO_HAVE_CXXABI_H
 	#endif
+	#if defined (__MINGW32__) || defined (__MINGW64__)
+		#define POCO_COMPILER_MINGW
+	#endif	
 #elif defined(_MSC_VER)
 	#define POCO_COMPILER_MSVC
 #elif defined (__GNUC__)
